Client/Frontend/Settings/LoginDetailViewController.swift

```diff
@@ -379,9 +379,15 @@ extension LoginDetailViewController {
             return
         }
 
-        let updated = Login(guid: login.guid, hostname: login.hostname, username: username, password: password)
-        if updated.isValid.isSuccess {
+        // Keep a copy of the old data in case we fail and need to revert back
+        let oldPassword = login.password
+        let oldUsername = login.username
+        login.updateWithPassword(password, username: username)
+
+        if login.isValid.isSuccess {
             profile.logins.updateLoginByGUID(login.guid, new: login, significant: true)
+        } else if let oldUsername = oldUsername {
+            login.updateWithPassword(oldPassword, username: oldUsername)
```

Comment:
But we're not redisplaying (undoing the user's edits), right? I'm not sure what the point is of flipping back.